ethcore/wasm/src/runtime.rs

```diff
@@ -530,7 +530,13 @@ impl<'a> Runtime<'a> {
 			* U256::from(self.ext.schedule().wasm().opcodes_mul)
 			/ U256::from(self.ext.schedule().wasm().opcodes_div);
 
-		match self.ext.create(&gas_left, &endowment, &code, &self.context.code_version, scheme, false).ok().expect("Trap is false; trap error will not happen; qed") {
+		let create_type = match scheme {
+			vm::CreateContractAddress::FromSenderAndNonce => ActionType::Create,
+			vm::CreateContractAddress::FromSenderSaltAndCodeHash(_) => ActionType::Create2,
+			vm::CreateContractAddress::FromSenderAndCodeHash => ActionType::Create,
```

Comment:
I think that should be `Create2` as well.

Reply:
Strictly speaking that's neither `CREATE` or `CREATE2` in EVM -- it's a custom address scheme used in pWasm.

Reply:
Yes, that's very true. I think it's based on some early proposal of `CREATE2` and mentally to me it's closer to `CREATE2` than `CREATE` (i.e. contract address is based on the code hash not account+nonce).